ipam/allocator.go

```diff
@@ -378,6 +381,36 @@ func (alloc *Allocator) ContainerStarted(ident string) {
 	}
 }
 
+func (alloc *Allocator) ContainerConnected(ident string) {
+	alloc.actionChan <- func() {
+		delete(alloc.disconnected, ident) // delete is no-op if key not in map
+	}
+}
+
+func (alloc *Allocator) ContainerDisconnected(ident string) {
+	alloc.actionChan <- func() {
+		if alloc.hasOwnedByContainer(ident) {
+			alloc.debugln("Container", ident, "disconnected; noting to remove later")
```

Comment:
We don't really want this because Docker tells us directly via the plugin when it is connecting or disconnecting a container.